tests/integration/kubernetes/gha-run.sh

```diff
@@ -34,16 +34,51 @@ function login_azure() {
 }
 
 function create_cluster() {
+    platform="${1:-}"
     # First, ensure that the cluster didn't fail to get cleaned up from a previous run.
-    delete_cluster || true
-
-    az aks create \
-        -g "${AZ_RG}" \
-        -n "$(_print_cluster_name)" \
-        -s "Standard_D4s_v5" \
-        --node-count 1 \
-        --generate-ssh-keys \
-        $([ "${KATA_HOST_OS}" = "cbl-mariner" ] && echo "--os-sku AzureLinux --workload-runtime KataMshvVmIsolation")
+    delete_cluster "${platform}"|| true
+
+    if [[ -z "$platform" || "$platform" = "aks" ]]; then
+        az aks create \
+            -g "${AZ_RG}" \
+            -n "$(_print_cluster_name)" \
+            -s "Standard_D4s_v5" \
+            --node-count 1 \
+            --generate-ssh-keys \
+            $([ "${KATA_HOST_OS}" = "cbl-mariner" ] && echo "--os-sku AzureLinux --workload-runtime KataMshvVmIsolation")
+    elif [ "$platform" = "kcli" ]; then
+        CLUSTER_NAME="${CLUSTER_NAME:-kata-k8s}"
+        kcli create kube "${KUBE_TYPE:-generic}" \
+            -P domain="kata.com" \
+            -P pool="${LIBVIRT_POOL:-default}" \
+            -P ctlplanes="${CLUSTER_CONTROL_NODES:-1}" \
+            -P workers="${CLUSTER_WORKERS:-1}" \
+            -P network="${LIBVIRT_NETWORK:-default}" \
+            -P image="${CLUSTER_IMAGE:-ubuntu2004}" \
+            -P sdn=flannel \
+            -P nfs=false \
+            -P disk_size="${CLUSTER_DISK_SIZE:-20}" \
```

Comment:
I'd also add `-P cpumodel=host-passthrough`, as this will avoid some issues when testing Cloud Hypervisor.